fetch/fetch.go

```diff
@@ -177,10 +177,14 @@ type network interface {
 
 // Fetch is the main struct that contains network peers and logic to batch and dispatch hash fetch requests
 type Fetch struct {
-	cfg                  Config
-	log                  log.Log
-	dbs                  map[Hint]database.Store
-	activeRequests       map[types.Hash32][]*request
+	cfg Config
+	log log.Log
+	dbs map[Hint]database.Store
+	// activeRequests contains requests that are not processed
+	activeRequests map[types.Hash32][]*request
+	// pendingRequests contains requests that have been processed and are waiting for responses
+	pendingRequests map[types.Hash32][]*request
```

Comment:
why add this distinction? an active request is a request that hasn't received a response yet. why do we need to distinct between active requests and requests waiting for response?

Reply:
because we don't delete the requests from activeRequests when we call requestHashBatchFromPeers(). so when we call requestHashBatchFromPeers() is called again before the first one got response back, the same set of hashes will be requested again.

you can see it in the test i added that SendRequest() is called twice for the same hash.